scripts.js

```diff
@@ -0,0 +1,423 @@
+		   'CapsLock', 'KeyA', 'KeyS', 'KeyD', 'KeyF', 'KeyG', 'KeyH', 'KeyJ', 'KeyK', 'KeyL', 'Semicolon', 'Quote', 'Enter',
+		   'ShiftLeft', 'KeyZ', 'KeyX', 'KeyC', 'KeyV', 'KeyB', 'KeyN', 'KeyM', 'Comma', 'Period', 'Slash', 'ArrowUp', 'ShiftRight',
+		   'ControlLeft', 'MetaLeft', 'AltLeft', 'Space', 'AltRight', 'ArrowLeft', 'ArrowDown', 'ArrowRight', 'ControlRight'
+		];
+
+const keysNum = [
+		layout.rus.default.regular[0].length, 
+		layout.rus.default.regular[1].length, 
+		layout.rus.default.regular[2].length, 
+		layout.rus.default.regular[3].length, 
+		layout.rus.default.regular[4].length
+		];
+
+class Keyboard {
+	constructor () {
+		if (localStorage.getItem('language') === null) {
+			this.language = navigator.language.includes('ru') ? 'rus' : 'eng';
+			localStorage.setItem('language', this.language);
+		}
+		else this.language = localStorage.getItem('language');
+		this.isShifted = false;
+		this.isCapsed = false;
+		this.option1 = this.isShifted ? 'shift' : 'default';
+		this.option2 = this.isCapsed ? 'caps' : 'regular';
+		this.layout = [...layout[this.language][this.option1][this.option2]];
+		this.keyboard = document.createElement('div');
+		this.keyboard.className = 'keyboard';
+		this.keys = [];
+	}
+
+	createLayout() {
+		this.layout.forEach((subArr, rowIdx) => {
+			let row = document.createElement('div');
+			row.className = 'row';
+			subArr.forEach((el, elIdx) => {
+				let key = document.createElement('div');
+				key.innerHTML = this.layout[rowIdx][elIdx];
+				key.className = 'key';
+				row.append(key);
+				this.keys.push(key);
+			});
+			this.keyboard.append(row);
+		})
+		document.body.append(this.keyboard);
+		return;
+	}
+
+	shift() {
+		this.isShifted = !this.isShifted;
+		this.changeLayout(this.language, this.isShifted, this.isCapsed);
+		return;
+	}
+
+	changeLanguage() {
+		this.language = this.language === 'rus' ? 'eng' : 'rus';
+		this.changeLayout(this.language, this.isShifted, this.isCapsed);
+		localStorage.setItem('language', this.language);
+	}
+
+	changeLayout(language, isShifted, isCapsed) {
+		this.isShifted = isShifted;
+		this.option1 = this.isShifted ? 'shift' : 'default';
+		this.isCapsed = isCapsed;
+		this.option2 = this.isCapsed ? 'caps' : 'regular';
+		this.language = language;
+		this.layout = [...layout[this.language][this.option1][this.option2]];
+		this.keys.forEach((key, i) => {
+			if (i < keysNum[0])	key.innerHTML = this.layout[0][i];
+			else if (i < keysNum[0] + keysNum[1]) {
+				key.innerHTML = this.layout[1][i - keysNum[0]];
+			} else if (i < keysNum[0] + keysNum[1] + keysNum[2]) {
+				key.innerHTML = this.layout[2][i - keysNum[0] - keysNum[1]];
+			}	else if (i < keysNum[0] + keysNum[1] + keysNum[2] + keysNum[3]) {
+				key.innerHTML = this.layout[3][i - keysNum[0] - keysNum[1] - keysNum[2]];
+			} else {
+				key.innerHTML = this.layout[4][i - keysNum[0] - keysNum[1] - keysNum[2] - keysNum[3]];
+			}
```

Comment:
I don't know how it works but it's definitely not a good way to code. Too difficult to grasp, too easy to make a mistake

Reply:
You could make this piece of code better by splitting it into a bunch of functions with proper names or by introducing additional boolean variables.
```javascript
const isPizzaBig = i < keysNum[0];
const isPizzaCheap = i < keysNum[0] + keysNum[1];
if (isPizzaBig)	
  key.innerHTML = this.layout[0][i];
else if (isPizzaCheap ) {
  key.innerHTML = this.layout[1][i - keysNum[0]];
}
```